hws/src/dr_remote.py

```diff
@@ -0,0 +1,133 @@
+#Copyright (c) 2025 NVIDIA CORPORATION. All rights reserved.
+
+import time
+import sys
+import os
+
+from src.dr_common import *
+from src.dr_common_functions import import_library
+from src.dr_db import _config_args
+
+SSH_OPTIONS = "-q -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oPubkeyAuthentication=no"
+
+def remote_cmd(cmd, host, user, password, time_out=60):
+    f = tempfile.mktemp()
+    fout = open(f, 'w')
+
+    cmd = "ssh %s@%s %s \"%s\"" % (user, host, SSH_OPTIONS, cmd)
+    p = pexpect.spawnu(cmd, timeout=time_out)
+
+    """
+    Expect remote response asking for password
+    """
+    i = p.expect(['[Yy]es', 'YES', '[Pp]assword: '])
+    if i < 2:
+        p.sendline('YES')
+        i = p.expect(['[Yy]es', 'YES', '[Pp]assword: '])
+        if i < 2:
+            p.sendline('Y')
+            i = p.expect(['[Yy]es', 'YES', '[Pp]assword: '])
+            if i < 2:
+                print("Connection error")
+                sys.exit(1)
+
+    p.sendline(password)
+    p.logfile = fout
+    p.expect(pexpect.EOF)
+    p.close()
+    fout.close()
+
+    fin = open(f, 'r')
+    stdout = fin.read()
+    fin.close()
+
+    if p.exitstatus != 0:
+        raise Exception(stdout)
+
+    return stdout
+
+
+def build_options():
+    op = ""
+
+    if _config_args.get("device") != None:
+        op += " -d %s " % (_config_args.get("device"))
+
+    if _config_args.get("verbose") > 0:
+        op += " -%s " % (_config_args.get("verbose") * "v")
+
+    if _config_args.get("statistics") == True:
+        op += " -s "
+
+    if _config_args.get("args.extra_hw_res") != "":
+        op += " --extra_hw_res %s " % _config_args.get("args.extra_hw_res")
+
+    if _config_args.get("parse_hw_resources") == False:
+        op += " --skip_parse "
+
+    return op
+
+
+def dr_connect_to_remote():
+    if _config_args.get("remote_dep_lib") == False:
+        globals()["pexpect"] = import_library("pexpect")
+        globals()["tempfile"] = import_library("tempfile")
+        globals()["getpass"] = import_library("getpass")
+        _config_args["remote_dep_lib"] = True
+
+    user_name = _config_args.get("user_name")
+    raw_dump_file = _config_args.get("file_path")
+    remote_ip = _config_args.get("remote_ip")
+    options = build_options()
+    remote_dump_file = "/tmp/dump_%s.csv" % (str(time.time()))
+    dtrp = _config_args.get("dump_tool_remote_path")#dump tool remote path, "" if not provided
+
+    password = getpass.getpass("Please enter %s@%s password:" % (user_name, remote_ip))
+    _config_args["password"] = password
+
+    print("connecting to remote...")
+
+    if dtrp == "":
+        #Check if mlx_steering_dump already exists in remote under /tmp/
+        cmd = "ls /tmp/"
+        out = remote_cmd(cmd, remote_ip, user_name, password, 15)
+
+        dtrp = "/tmp/mlx_steering_dump"
+
+        if "mlx_steering_dump" not in out.split("\n"):
+            print("Cloning mlx steering dump tool to remote")
```

Comment:
please replace "cloning" with "copying" - cloning implies 'git clone', we're not doing that